src/main/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -98,68 +108,186 @@ public SAMSequenceRecord(final String name, final int sequenceLength) {
+        return (anTag == null) ? Collections.emptySet()
+                : Collections.unmodifiableSet(new LinkedHashSet<>(Arrays.asList(anTag.split(ALTERNATIVE_SEQUENCE_NAME_SEPARATOR))));
+    }
+
+    /**
+     * Adds an alternative sequence name if it is not the same as the sequence name or it is not present already.
+     */
+    public void addAlternativeSequenceName(final String name) {
+        validateAltRegExp(name);
+        final Set<String> altSequences = new HashSet<>(getAlternativeSequenceNames());
+
+        if (!mSequenceName.equals(name)) {
+            altSequences.add(name);
+        }
+        encodeAltSequences(altSequences);
+    }
+
+    /**
+     * Sets the alternative sequence names in the order provided by iteration, removing the previous values.
+     */
+    public void setAlternativeSequenceName(final Collection<String> alternativeSequences) {
+        if (alternativeSequences == null) {
+            setAttribute(ALTERNATIVE_SEQUENCE_NAME_TAG, null);
+        } else {
+            // validate all the names and encode afterwards
+            alternativeSequences.forEach(SAMSequenceRecord::validateAltRegExp);
+            encodeAltSequences(alternativeSequences);
+        }
+    }
+
+    private static void validateAltRegExp(final String name) {
+        if (!LEGAL_RNAME_PATTERN.matcher(name).matches()) {
+            throw new IllegalArgumentException(String.format("Invalid alternative sequence name '%s': do not match the pattern %s", name, LEGAL_RNAME_PATTERN));
+        }
+    }
+
+    private void encodeAltSequences(final Collection<String> alternativeSequences) {
+
+        //make sure that the order in which alternate names are joined is determined
+        setAttribute(ALTERNATIVE_SEQUENCE_NAME_TAG, alternativeSequences.isEmpty() ? null : alternativeSequences.stream()
+                .sorted()
+                .distinct()
+                .peek(SAMSequenceRecord::validateAltRegExp)
+                .collect(Collectors.joining(ALTERNATIVE_SEQUENCE_NAME_SEPARATOR)));
+    }
+
+    /**
+     * Returns {@code true} if there are alternative sequence names; {@code false} otherwise.
+     */
+    public boolean hasAlternativeSequenceNames() {
+        return getAttribute(ALTERNATIVE_SEQUENCE_NAME_TAG) != null;
+    }
 
     /**
      * Looser comparison than equals().  We look only at sequence index, sequence length, and MD5 tag value
      * (or sequence names, if there is no MD5 tag in either record.
      */
     public boolean isSameSequence(final SAMSequenceRecord that) {
-        if (this == that) return true;
-        if (that == null) return false;
+        if (this == that) {
+            return true;
+        }
+        if (that == null) {
+            return false;
+        }
 
-        if (mSequenceIndex != that.mSequenceIndex) return false;
+        if (mSequenceIndex != that.mSequenceIndex) {
+            return false;
+        }
         // PIC-439.  Allow undefined length.
```

Comment:
Do we know if this is still an issue?

Reply:
Not sure what the numbering scheme refers to….and I can't find it online either….however, it does not pertain to this PR…